src/Marten/Events/Projections/MultiStreamProjection.cs

```diff
@@ -108,6 +108,21 @@ protected override void specialAssertValid()
         _defaultSlicer.FanOut(fanOutFunc, mode);
     }
 
+    /// <summary>
+    ///     Apply "fan out" operations to the given IEvent<TEvent> type that inserts an enumerable of TChild events right behind the
+    ///     parent
+    ///     event in the event stream
+    /// </summary>
+    /// <param name="fanOutFunc"></param>
+    /// <param name="mode">Should the fan out operation happen after grouping, or before? Default is after</param>
+    /// <typeparam name="TEvent"></typeparam>
+    /// <typeparam name="TChild"></typeparam>
+    public void FanOutEvent<TEvent, TChild>(Func<IEvent<TEvent>, IEnumerable<TChild>> fanOutFunc,
```

Comment:
**SUGGESTION:** My preference would be to name it as `FanOut` and just have overload instead of name with the suffix
```suggestion
    public void FanOut<TEvent, TChild>(Func<IEvent<TEvent>, IEnumerable<TChild>> fanOutFunc,
```

Reply:
hmm, I do not understand how it should know what overload to pick, but it seems to work now. I thought it did not before at some time during development.

Reply:
@erdtsieck .NET compiler is usually smart enough to select the best matching overload. So, as the new one has clear restriction on IEvent, then it should find it based on the criteria. That doesn't work if this would be an extension method, as regular methods have precedence over them.

Reply:
@oskardudycz I do understand that it picks the best overload, but since I call it from the example on a type that is not IEvent, I can not imagine how it should know.

```cs
FanOut<Travel, Stop>(x => x.Data.Stops);
```

Maybe because i try to access `.Data`, but that seems far fetched